Paintroid/src/main/java/org/catrobat/paintroid/tools/helper/JavaCropAlgorithm.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Paintroid: An image manipulation application for Android.
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.paintroid.tools.helper
+
+import android.graphics.Bitmap
+import android.graphics.Color
+import android.graphics.Rect
+import android.util.Log
+
+private const val TAG = "cropAlgorithmSnail"
+
+class JavaCropAlgorithm : CropAlgorithm {
+    private fun getRange(s: Int, e: Int): ArrayList<Int> {
+        val range = ArrayList<Int>()
+        for (i in s..e) range.add(i)
+        return range
+    }
+
+    private fun getTopmostCellWithOpaquePixel(bounds: Rect, pixels: Array<IntArray>): Int? =
+        getRange(bounds.top, bounds.bottom).firstOrNull { row ->
+            getRange(
+                bounds.left,
+                bounds.right
+            ).firstOrNull { col -> pixels[row][col] != Color.TRANSPARENT } != null
+        }
+
+    private fun getBottommostCellWithOpaquePixel(bounds: Rect, pixels: Array<IntArray>): Int? =
+        getRange(bounds.top, bounds.bottom).lastOrNull { row ->
+            getRange(
+                bounds.left,
+                bounds.right
+            ).firstOrNull { col -> pixels[row][col] != Color.TRANSPARENT } != null
+        }
+
+    private fun getLeftmostCellWithOpaquePixel(bounds: Rect, pixels: Array<IntArray>): Int? =
+        getRange(bounds.left, bounds.right).firstOrNull { col ->
+            getRange(
+                bounds.top,
+                bounds.bottom
+            ).firstOrNull { row -> pixels[row][col] != Color.TRANSPARENT } != null
```

Comment:
Instead of creating an Array<Array> we could simply use the bitmap and [getPixel](https://developer.android.com/reference/kotlin/android/graphics/Bitmap#getpixel)

Reply:
done :)